lib/orderbook/OrderBook.ts

```diff
@@ -277,6 +277,9 @@ class OrderBook extends EventEmitter {
         remainingOrder: order,
       });
     }
+    // TODO: Check channel balance to verify that we can swap the order amount.
+    // Case channel not existent -> error!
+    // Case channel with sufficient capacity exists, but not enough local balance -> warning!
```

Comment:
This comment was a result of the issue description in: https://github.com/ExchangeUnion/xud/issues/240

That was my reasoning as well - it doesn't make sense to have it in the `OrderBook` for now. The only reason we'd want to have an additional check here is to prevent spoofing orders that can/will never be swapped.

Reply:
Ah my mistake I had not looked at that issue in a while. I commented there on whether we should actually require a direct channel, but either way I don't think we'd have it in OrderBook.ts.